### PR TITLE
Mosip-34016 Merge java 21 changes to develop

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: authentication-demo-service/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 	
 	<groupId>io.mosip.testrig.authentication.demo</groupId>
-	<version>1.2.0.1-SNAPSHOT</version>
+	<version>1.2.1-java21-SNAPSHOT</version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml:7]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml [hunk 2]
```diff
@@ -62,18 +62,18 @@
...
+		<maven.compiler.source>21</maven.compiler.source>
+		<maven.compiler.target>21</maven.compiler.target>
 		<maven.compiler.version>3.8.0</maven.compiler.version>
 		<maven.javadoc.version>3.0.1</maven.javadoc.version>
 		<maven.swagger.version>2.9.2</maven.swagger.version>
 		<maven.jacoco.version>0.8.2</maven.jacoco.version>
 		<sonar.skip>true</sonar.skip>
 		
-		<kernel-websubclient-api.version>1.2.0.1-SNAPSHOT</kernel-websubclient-api.version>
+		<kernel-websubclient-api.version>1.2.1-java21-SNAPSHOT</kernel-websubclient-api.version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml:76]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml [hunk 3]
```diff
@@ -112,13 +99,12 @@
...
 			<groupId>org.glassfish.jersey.core</groupId>
 			<artifactId>jersey-common</artifactId>
-			<version>2.22.2</version>
 			<scope>test</scope>
 		</dependency>
 		<dependency>
 			<groupId>io.mosip.authentication</groupId>
 			<artifactId>authentication-core</artifactId>
-			<version>1.2.0.1-B3</version>
+			<version>1.2.1-java21-SNAPSHOT</version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml:107]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml [hunk 4]
```diff
@@ -129,19 +115,17 @@
...
 			<groupId>org.mockito</groupId>
 			<artifactId>mockito-core</artifactId>
-			<version>2.23.4</version>
 		</dependency>
 
 		<dependency>
 			<groupId>io.mosip.kernel</groupId>
 			<artifactId>kernel-core</artifactId>
-			<version>1.2.0.1-SNAPSHOT</version>
+			<version>1.2.1-java21-SNAPSHOT</version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml:128]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml [hunk 5]
```diff
@@ -221,22 +202,19 @@
 		<dependency>
 			<groupId>commons-codec</groupId>
 		  	<artifactId>commons-codec</artifactId>
-		  	<version>1.11</version>
 		</dependency>
 		<dependency>
 			<groupId>io.mosip.kernel</groupId>
 			<artifactId>kernel-templatemanager-velocity</artifactId>
-			<version>1.2.0.1-SNAPSHOT</version>
+			<version>1.2.1-java21-SNAPSHOT</version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml:209]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml [hunk 6]
```diff
@@ -267,7 +245,27 @@
...
+			<artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+			<version>2.5.0</version>
+		</dependency>
 	</dependencies>
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<groupId>io.mosip.kernel</groupId>
+				<artifactId>kernel-bom</artifactId>
+				<version>1.2.1-java21-SNAPSHOT</version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml [hunk 7]
```diff
@@ -251,7 +229,7 @@
 		<dependency>
 			<groupId>io.mosip.kernel</groupId>
 			<artifactId>kernel-keymanager-service</artifactId>
-			<version>1.2.0.1-SNAPSHOT</version>
+			<version>1.2.1-java21-SNAPSHOT</version>
```
Comment (sowmya695) [inline authentication-demo-service/pom.xml]: If it is develop branch then version should be 1.2.1-SNAPSHOT

File: authentication-demo-service/pom.xml (hunk 1 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done.

File: authentication-demo-service/pom.xml (hunk 2 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done

File: authentication-demo-service/pom.xml (hunk 3 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done.

File: authentication-demo-service/pom.xml (hunk 4 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done.

File: authentication-demo-service/pom.xml (hunk 5 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done.

File: authentication-demo-service/pom.xml (hunk 6 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done.

File: authentication-demo-service/pom.xml (hunk 7 above)
Comment (kameshsr) [reply authentication-demo-service/pom.xml]: Done.

**Review by sowmya695 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**